packages/event-handler/src/rest/RouteHandlerRegistry.ts

```diff
@@ -0,0 +1,53 @@
+import type { GenericLogger } from '@aws-lambda-powertools/commons/types';
+import type { RouteRegistryOptions } from '../types/rest.js';
+import type { Route } from './Route.js';
+import { validatePathPattern } from './utils.js';
+
+class RouteHandlerRegistry {
+  readonly #routes: Map<string, Route> = new Map();
+  readonly #routesByMethod: Map<string, Route[]> = new Map();
+
+  readonly #logger: Pick<GenericLogger, 'debug' | 'warn' | 'error'>;
+
+  constructor(options: RouteRegistryOptions) {
+    this.#logger = options.logger;
+  }
+
+  public register(route: Route): void {
+    const { isValid, issues } = validatePathPattern(route.path);
+    if (!isValid) {
+      for (const issue of issues) {
+        this.#logger.warn(issue);
```

Comment:
I'm not sure about this, should we emit each issue in a single log entry/line or a single one with formatting?

Reply:
So are we thinking something like:

```ts
this.#logger.warn(`Route ${route.path} is invalid: \n ${issues.join('\n')}`);
```

Reply:
Yeah, I'm in two minds myself. Let's postpone our decision for now and see where we are when the handler is more fleshed out.